openedx/core/djangoapps/programs/utils.py

```diff
@@ -49,49 +46,25 @@ def get_programs(user, program_id=None):
         list of dict, representing programs returned by the Programs service.
         dict, if a specific program is requested.
     """
-    programs_config = ProgramsApiConfig.current()
-
-    # Bypass caching for staff users, who may be creating Programs and want
-    # to see them displayed immediately.
-    cache_key = programs_config.CACHE_KEY if programs_config.is_cache_enabled and not user.is_staff else None
-
-    programs = get_edx_api_data(programs_config, user, 'programs', resource_id=program_id, cache_key=cache_key)
-
-    # Mix in munged MicroMasters data from the catalog.
-    if not program_id:
-        programs += [
-            munge_catalog_program(micromaster) for micromaster in get_catalog_programs(user, type='MicroMasters')
-        ]
 
-    return programs
+    if use_catalog:
+        programs = [munge_catalog_program(program) for program in get_catalog_programs(user)]
```

Comment:
When you say combine is the idea that we look at the query params and use PageNumberPagination if the query has page/page_size params and limitoffset if it has limit/offset params?